anchore_engine/services/catalog/api/controllers/default_controller.py

```diff
@@ -98,7 +99,10 @@ def add_image(image_metadata=None, tag=None, digest=None, created_at=None, from_
 
     except Exception as err:
         logger.exception('Error processing image add')
-        httpcode = 500
+        if isinstance(err, ImageConflict):
```

Comment:
this could add another `try` block catching `ImageConflict` instead of inspecting the class hierarchy:
 ```suggestion
    except ImageConflict:
```

Although... because the previous `except Exception` comes first... then this would not be possible. This is yet another issue why overusing `except Exception` is not a good idea :(